spec/models/equipment_model_spec.rb

```diff
@@ -2,437 +2,343 @@
+    describe 'max_per_user' do
+      it_behaves_like 'integer attribute', :max_per_user
+      it_behaves_like 'does not allow 0', :max_per_user
     end
-
-    # max_renewal_times
-    it 'requires an integer value for maximum renewal times' do
-      @model.max_renewal_times = 2.3
-      expect(@model.save).to be_falsey
-      @model.max_renewal_times = 2
-      expect(@model.save).to be_truthy
-    end
-    it 'requires a maximum renewal times value greater than or equal to 0' do
-      @model.max_renewal_times = -1
-      expect(@model.save).to be_falsey
-      @model.max_renewal_times = 0
-      expect(@model.save).to be_truthy
-    end
-    it 'allows nil values for maximum renewal times' do
-      @model.max_renewal_times = nil
-      expect(@model.save).to be_truthy
+    describe 'max_renewal_length' do
+      it_behaves_like 'integer attribute', :max_renewal_length
+      it_behaves_like 'allows 0', :max_renewal_length
     end
-
-    # renewal_days_before_due
-    it 'requires an integer value for renewal days before due' do
-      @model.renewal_days_before_due = 2.3
-      expect(@model.save).to be_falsey
-      @model.renewal_days_before_due = 2
-      expect(@model.save).to be_truthy
-    end
-    it 'requires a renewal days before due value greater than or equal to 0' do
-      @model.renewal_days_before_due = -1
-      expect(@model.save).to be_falsey
-      @model.renewal_days_before_due = 0
-      expect(@model.save).to be_truthy
-    end
-    it 'allows nil values for renewal days before due' do
-      @model.renewal_days_before_due = nil
-      expect(@model.save).to be_truthy
+    describe 'max_renewal_times' do
+      it_behaves_like 'integer attribute', :max_renewal_times
+      it_behaves_like 'allows 0', :max_renewal_times
+    end
+    describe 'renewal_days_before_due' do
+      it_behaves_like 'integer attribute', :renewal_days_before_due
+      it_behaves_like 'allows 0', :renewal_days_before_due
+    end
+    shared_examples 'string attribute' do |attr|
+      it 'fails when less than 0' do
+        model = mock_eq_model(attr => '-1.00')
+        expect(model.valid?).to be_falsey
+      end
+      it 'can be 0' do
+        model = mock_eq_model(attr => '0.00')
+        expect(model.valid?).to be_truthy
+      end
+    end
+    describe 'late fee' do
+      it_behaves_like 'string attribute', :late_fee
+    end
+    describe 'replacement fee' do
+      it_behaves_like 'string attribute', :replacement_fee
     end
   end
 
-  context 'association validations' do
-    before(:each) do
-      @unique_id = FactoryGirl.generate(:unique_id)
-      @model = FactoryGirl.create(:equipment_model, id: @unique_id)
-    end
+  describe 'association validations' do
+    let!(:unique_id) { FactoryGirl.generate(:unique_id) }
     it 'has a working association callback' do
-      expect(@model.save).to be_truthy
+      model = mock_eq_model(id: unique_id)
+      expect(model.valid?).to be_truthy
```

Comment:
If we can't figure out what it's actually doing I feel like we're better off not testing it...